packages/agent-base/src/index.ts

```diff
@@ -99,6 +115,20 @@ export abstract class Agent extends http.Agent {
 			}, cb);
 	}
 
+	upgradeSocketToTls(
+		servername: string | undefined,
+		opts: tls.ConnectionOptions,
+		socket?: net.Socket,
```

Comment:
Seems weird to me that `socket` is optional. Also, let's make it the first parameter:

```suggestion
		socket: net.Socket,
		servername: string | undefined,
		opts: tls.ConnectionOptions,
```

Reply:
The reason it was optional was that pac-proxy-agent calls tls.connect without providing a socket. I've made your suggested changes and reverted the changes to pac-proxy-agent for now. Let me know how you would like to proceed in that regard.

Reply:
I see. How about if we do something like this in pac-proxy-agent?

```typescript
				socket = net.connect(opts);
				
				if (secureEndpoint) {
					const servername = opts.servername || opts.host;
					socket = this.upgradeSocketToTls(
						socket,
						servername,
						opts
					);
				}
```

Reply:
@mg-dd Did this piece of feedback get addressed? I don't see any change in pac-proxy-agent so it looks like this still needs to be done?